src/main/java/seedu/address/model/ContactTagIntegrationManager.java

```diff
@@ -0,0 +1,188 @@
+        tagTree.removeSubTagsFrom(superTag, subTagsToRemove);
+        tagTree.addSubTagsTo(superTag, subTagsToAdd);
+    }
+
+    /**
+     * Deletes a {@code tag} from the existing tag tree.
+     * {@code Person}s with the deleted {@code tag} will have the {@code tag} removed.
+     *
+     * @param tag {@code tag} to be deleted.
+     */
+    public void deleteTag(Tag tag) {
+        assert tag != null;
+
+        tagTree.deleteTag(tag);
+        removeTagFromContactsInAddressBook(tag);
+    }
+
+    /**
+     * Replaces each instance of a {@code person} with {@code tag} with a {@code person} without it.
+     */
+    private void removeTagFromContactsInAddressBook(Tag tag) {
+        Set<Person> setCopy = Set.copyOf(addressBook.getPersonsWithTag(tag));
+        setCopy.stream()
+                .forEach(person -> addressBook.setPerson(person, copyPersonWithoutTag(person, tag)));
+    }
+
+    /**
+     * Deletes {@code tag} and all its sub-{@code tag}s directly below {@code tag} in the tag hierarchy.
+     * For example, if {@code CS2030 -> COMPUTING -> NUS}, deleting NUS with this method
+     * will delete {@code CS2030} and {@code COMPUTING}.
+     * Affected {@code Person}s will be updated accordingly.
+     *
+     * @param tag {@code tag} to begin the deletion.
+     */
+    public void deleteTagRecursive(Tag tag) {
+        removeTagFromContactsInAddressBook(tag);
+        tagTree.getSubTagsOf(tag).stream().forEach(subTag -> deleteTagRecursive(subTag));
+        tagTree.deleteTagAndAllSubTags(tag);
+    }
+
+    /**
+     * Deletes {@code tag} and all {@code person} objects directly assigned to it.
+     *
+     * @param tag {@code tag} to be deleted together with the {@code person}s directly under this {@code tag}.
+     */
+    public void deleteTagAndDirectContacts(Tag tag) {
+        tagTree.deleteTag(tag);
+        removeContactsUnderTagFromAddressBook(tag);
+    }
+
+    /**
+     * Removes all {@code person}s directly under {@code tag} from the {@code addressbook}.
+     */
+    private void removeContactsUnderTagFromAddressBook(Tag tag) {
+        Set<Person> setCopy = Set.copyOf(addressBook.getPersonsWithTag(tag));
+        setCopy.stream()
+                .forEach(person -> addressBook.removePerson(person));
+    }
+
+    /**
+     * Deletes {@code tag} and all its sub-{@code tag}s directly below {@code tag} in the tag hierarchy.
+     * All {@code person}s under any deleted {@code tag} will also be deleted.
+     *
+     * @param tag {@code tag} to begin recursive deletion.
+     */
+    public void deleteTagAndDirectContactsRecursive(Tag tag) {
+        removeContactsUnderTagFromAddressBook(tag);
+        tagTree.getSubTagsOf(tag).stream().forEach(subTag -> deleteTagAndDirectContactsRecursive(subTag));
+        tagTree.deleteTagAndAllSubTags(tag);
+    }
+
+    /**
+     * returns a new {@code person} object which is nearly identical to {@code persontocopy} except with
+     * {@code tagRemoved} removed.
```

Comment:
lacking capitalisation; also i think it should be `personToCopy` rather than `persontocopy`.